op-challenger/cmd/create_game.go

```diff
@@ -0,0 +1,137 @@
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "OUTPUT_ROOT"),
+	}
+	L2BlockNumFlag = &cli.StringFlag{
+		Name:    "l2-block-num",
+		Usage:   "The l2 block number for the game.",
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "L2_BLOCK_NUM"),
+	}
+)
+
+func CreateGame(ctx *cli.Context) error {
+	logger, err := setupLogging(ctx)
+	if err != nil {
+		return err
+	}
+	rpcUrl := ctx.String(flags.L1EthRpcFlag.Name)
+	if rpcUrl == "" {
+		return fmt.Errorf("missing %v", flags.L1EthRpcFlag.Name)
+	}
+	gameAddr, err := opservice.ParseAddress(ctx.String(GameAddressFlag.Name))
+	if err != nil {
+		return err
+	}
+	outputRoot, err := opservice.ParseAddress(ctx.String(OutputRootFlag.Name))
+	if err != nil {
+		return err
+	}
+	traceType := ctx.Uint64(TraceTypeFlag.Name)
+	l2BlockNum := ctx.Uint64(L2BlockNumFlag.Name)
+
+	l1Client, err := dial.DialEthClientWithTimeout(ctx.Context, dial.DefaultDialTimeout, logger, rpcUrl)
+	if err != nil {
+		return fmt.Errorf("failed to dial L1: %w", err)
+	}
+	defer l1Client.Close()
+
+	caller := batching.NewMultiCaller(l1Client.Client(), batching.DefaultBatchSize)
+	contract, err := contracts.NewFaultDisputeGameContract(gameAddr, caller)
+	if err != nil {
+		return fmt.Errorf("failed to create dispute game bindings: %w", err)
+	}
+
+	txMgrConfig := txmgr.ReadCLIConfig(ctx)
+	txMgr, err := txmgr.NewSimpleTxManager("challenger", logger, &metrics.NoopTxMetrics{}, txMgrConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create the transaction manager: %w", err)
+	}
+
+	inner := contract.Inner()
+	methodCreateGame := "create"
+
+	call := inner.Call(methodCreateGame, uint32(traceType), outputRoot, l2BlockNum)
+	_, err = caller.SingleCall(context.Background(), batching.BlockLatest, call)
+	if err != nil {
+		return fmt.Errorf("failed to call create: %w", err)
+	}
+
+	call = inner.Call(methodCreateGame, uint32(traceType), outputRoot, l2BlockNum)
+	txCandidate, err := call.ToTxCandidate()
+	if err != nil {
+		return fmt.Errorf("failed to create tx candidate: %w", err)
+	}
+
+	rct, err := txMgr.Send(context.Background(), txCandidate)
+	if err != nil {
+		return fmt.Errorf("failed to send tx: %w", err)
+	}
+	fmt.Printf("Sent create transaction with status %v, tx_hash: %s\n", rct.Status, rct.TxHash.String())
+
+	call = inner.Call("getGameUUID", traceType, outputRoot, l2BlockNum)
+	uuid, err := caller.SingleCall(context.Background(), batching.BlockLatest, call)
+	if err != nil {
+		return fmt.Errorf("failed to call getGameUUID: %w", err)
+	}
+	gameUUID := uuid.GetHash(0)
+	fmt.Printf("Calculated Game UUID: %s\n", gameUUID.String())
+
+	call = inner.Call("games", traceType, outputRoot, l2BlockNum)
+	fetchedGameAddr, err := caller.SingleCall(context.Background(), batching.BlockLatest, call)
+	if err != nil {
+		return fmt.Errorf("failed to call games: %w", err)
+	}
+	parsedGameAddr := fetchedGameAddr.GetAddress(0)
+	fmt.Printf("Fetched Game Address: %s\n", parsedGameAddr.String())
```

Comment:
We could extract this from the transaction receipt. Parsing from the receipt is probably ideal, but I think there's likely to be a bit more framework code to manage to handle decoding logs since we haven't done that yet. Suspect it would be useful to have but just looking up the game data via the eth_call works so happy to go whichever way you think.

Either way the code to get the address should move into the contract bindings.

Reply:
Moved into the contract bindings for now. I'd like to avoid receipt parsing in this pr and make that optimization for core challenger behavior as opposed to these test subcommands if possible.